test/public/stageView.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { _uninitialize, _initialize } from '../../src/public/publicAPIs';
+import { Utils } from '../utils';
+import { stageView } from '../../src/public';
+
+describe('stageView', () => {
+  const utils = new Utils();
+
+  beforeEach(() => {
+    utils.processMessage = null;
+    utils.messages = [];
+    utils.childMessages = [];
+    utils.childWindow.closed = false;
+
+    // Set a mock window for testing
+    _initialize(utils.mockWindow);
+  });
+
+  afterEach(() => {
+    // Reset the object since it's a singleton
+    if (_uninitialize) {
+      _uninitialize();
+    }
+  });
+
+  const stageViewParams: stageView.StageViewParams = {
+    appId: 'appId',
+    contentUrl: 'contentUrl',
+    threadId: 'threadId',
+    title: 'title',
+    websiteUrl: 'websiteUrl',
+    entityId: 'entityId',
+  };
+
+  describe('open', () => {
+    it('should not allow calls before initialization', () => {
+      expect(() => stageView.open(stageViewParams)).toThrowError('The library has not yet been initialized');
+    });
+
+    it('should not allow calls from stage context', () => {
```

Comment:
Can you unit test the other frame contexts that are not supported as well? Should be pretty straightforward to write as a for loop, let me know if you have trouble.